systems/src/apu/mod.rs

```diff
@@ -0,0 +1,1396 @@
+            let mut tester = tester_with().starting_apu();
+
+            loop {
+                tester = tester.run(Duration::from_secs(1));
+
+                if tester.apu_is_available() {
+                    break;
+                }
+            }
+
+            assert!(!tester.start_is_on());
+            assert!(tester.start_shows_available());
+        }
+
+        #[test]
+        fn when_apu_bleed_valve_open_on_shutdown_cooldown_period_commences_and_apu_remains_available(
+        ) {
+            // The cool down period is between 60 to 120. It is configurable by aircraft mechanics and
+            // we'll make it a configurable option in the sim. For now, 120s.
+            let mut tester =
+                tester_with()
+                    .running_apu()
+                    .and()
+                    .master_off()
+                    .run(Duration::from_millis(
+                        ElectronicControlBox::BLEED_AIR_COOLDOWN_DURATION_MILLIS,
+                    ));
+
+            assert!(tester.apu_is_available());
+
+            // Move from Running to Shutdown turbine state.
+            tester = tester.run(Duration::from_millis(1));
+            // APU N reduces below 95%.
+            tester = tester.run(Duration::from_secs(5));
+
+            assert!(!tester.apu_is_available());
+        }
+
+        #[test]
+        fn when_shutting_down_apu_remains_available_until_n_less_than_95() {
+            let mut tester = tester_with()
+                .running_apu_without_bleed_air()
+                .and()
+                .master_off();
+
+            let mut n = 100.;
+
+            while 0. < n {
+                tester = tester.run(Duration::from_millis(50));
+                n = tester.get_n().get::<percent>();
+                assert_eq!(tester.apu_is_available(), 95. <= n);
+            }
+        }
+
+        #[test]
+        fn when_apu_bleed_valve_was_open_recently_on_shutdown_cooldown_period_commences_and_apu_remains_available(
+        ) {
+            // The cool down period requires that the bleed valve is shut for a duration (default 120s).
+            // If the bleed valve was shut earlier than the MASTER SW going to OFF, that time period counts towards the cool down period.
+
+            let mut tester = tester_with()
+                .running_apu_with_bleed_air()
+                .and()
+                .bleed_air_off()
+                .run(Duration::from_millis(
+                    (ElectronicControlBox::BLEED_AIR_COOLDOWN_DURATION_MILLIS / 3) * 2,
+                ));
+
+            assert!(tester.apu_is_available());
+
+            tester = tester.master_off().run(Duration::from_millis(
+                ElectronicControlBox::BLEED_AIR_COOLDOWN_DURATION_MILLIS / 3,
+            ));
+
+            assert!(tester.apu_is_available());
+
+            // Move from Running to Shutdown turbine state.
+            tester = tester.run(Duration::from_millis(1));
+            // APU N reduces below 95%.
```

Comment:
Could assert this to make the test more resilient to timing changes (a failing assertion would help a developer find the error more quickly).